lib/rubocop/cop/lint/meterpreter_commands_dependencies.rb

```diff
@@ -0,0 +1,260 @@
+                "#{commands_whitespace}]\n" \
+                "#{meterpreter_whitespace}}\n" \
+                "#{compat_whitespace}}"
+
+              corrector.replace(compat_hash_node, new_hash)
+
+            elsif @compat_node.nil? && @meterpreter_node.nil? && @command_node.nil? && !@initialize_node.nil?
+              # require "pry"; binding.pry
+              # White spacing handling based of node offsets
+              compat_whitespace = offset(@end_of_info_node)
+              meterpreter_whitespace = compat_whitespace + "  "
+              commands_whitespace = meterpreter_whitespace + "  "
+              array_content_whitespace = commands_whitespace + "  "
+
+              # Formatting to add missing commands node when the method has a compat node & meterpreter node present
+              test_new_hash =
+                ",\n#{compat_whitespace}'Compat' => {\n" \
+                "#{meterpreter_whitespace}'Meterpreter' => {\n" \
+                "#{commands_whitespace}'Commands' => %w[" \
+                "\n#{array_content_whitespace}#{@identified_commands.join("\n#{array_content_whitespace}")}\n" \
+                "#{commands_whitespace}]\n" \
+                "#{meterpreter_whitespace}}\n" \
+                "#{compat_whitespace}}"
+
+              corrector.insert_after(@end_of_info_node, test_new_hash)
+
+            elsif @compat_node.nil? && @meterpreter_node.nil? && @command_node.nil? && @initialize_node.nil?
+              # White spacing handling based of node offsets
+              def_whitespace = offset(@node_after_initialize)
+              super_whitespace = def_whitespace + "  "
+              update_info_whitespace = super_whitespace + "  "
+              info_whitespace = update_info_whitespace + "  "
+              meterpreter_whitespace = info_whitespace + "  "
+              commands_whitespace = meterpreter_whitespace + "  "
+              array_content_whitespace = commands_whitespace + "  "
+
+              # Formatting to add missing commands node when the method has a compat node & meterpreter node present
+              new_hash =
+                "def initialize(info = {})" \
+                "\n#{super_whitespace}super(" \
+                "\n#{update_info_whitespace}update_info(" \
+                "\n#{info_whitespace}info," \
+                "\n#{info_whitespace}'Compat' => {" \
+                "\n#{meterpreter_whitespace}'Meterpreter' => {" \
+                "\n#{commands_whitespace}'Commands' => %w[" \
+                "\n#{array_content_whitespace}#{@identified_commands.join("\n#{array_content_whitespace}")}" \
+                "\n#{commands_whitespace}]" \
+                "\n#{meterpreter_whitespace}}" \
+                "\n#{info_whitespace}}" \
+                "\n#{update_info_whitespace})" \
+                "\n#{super_whitespace})" \
+                "\n#{def_whitespace}end" \
+                "\n  "
+
```

Comment:
We'll want to DRY this all out, either via more clever string manipulation, or a potential alternative algorithm might be:

1. Generate the template which would have everything
```ruby
# (Formatted correctly)
def generate_initialize_template
    <<~EOF
       def initialize(...)
           super(
               update_info(
                  info,
                   {
                    compat => meterpreter => commands [
                      ...
                     ]
                 }
             )
           )
       end
    EOF
end
```
1. Parse the template again with rubocop:
```
source = RuboCop::ProcessedSource.new(code, RUBY_VERSION.to_f)
node = source.ast
```
1. Decide which nodes are present in the original source, then extract the subset of the 'everything' template that you need, and append it to the right place in the original source. Just using normal rubocop node matchers to extract the data out. Would need to ensure it's indented to the right level, but that should be pretty trivial